custom-plugin-decorators/tag-sorting/README.md

```diff
@@ -0,0 +1,95 @@
+# Decorator to sort tags in an API description
+
+Authors:
+- [`@lornajane`](https://github.com/lornajane), Lorna Mitchell (Redocly)
+ 
+## What this does and why
+
+Redocly CLI has the [`tags-alphabetical'](https://redocly.com/docs/cli/rules/tags-alphabetical/) rule to error if the `tags` section isn't in alphabetical order by `name`.
+
+This decorator provides functionality to _put_ the tags in order by name, in order to make any API description compatible with that rule.
+
+## Code
+
+Here's the main plugin entrypoint, it's in `tag-sorting.js`:
+
+```javascript
+
+const SortTagsAlphabetically = require('./decorator-alpha.js');
+
+module.exports = {
+  id: 'tag-sorting',
+  decorators: {
+    oas3: {
+      'alphabetical': SortTagsAlphabetically,
+    }
+  }
+	
+}
+```
+
+The code here sets the plugin name to `tag-sorting` and adds a decorator for OpenAPI 3 called `alphabetical`.
+
+The functionality for the alphabetical sorting is in `decorator-alpha.js`:
+
+```javascript
+
+module.exports = SortTagsAlphabetically;
+
+function SortTagsAlphabetically() {
+  console.log("re-ordering tags: alphabetical");
+  return {
+    TagList: {
+      leave(target) {
+        target.sort((a,b) => {
```

Comment:
We added a flag/option to make it case insensitive for sorting purposes (which is more similar to how humans sort). What do you think about adding that here too?

Reply:
+1 on this idea. It would also give users an example of how they can pass flags, too.

Reply:
Alternatively, I think defaulting to case insensitive would be reasonable behavior.

Reply:
Not blocking, but maybe we can add it as an issue in case someone else wants to add it later.

Reply:
I have a different perspective about case-sensitive being the expected behavior. I don't think this is too important because these examples in the cookbook don't have to be practical. 

If you were to search for "API keys" in these knowing that it was sorted alphabetically, which order would you expect them in?

Case-insensitive sort order:

- Accounts
- Allowlists
- AML
- API keys
- Application owners
- Application users
- Coupons

Case-sensitive sort order:

- AML
- API keys
- Accounts
- Allowlists
- Application owners
- Application users
- Coupons

Reply:
That's a really great example. Thanks, @adamaltman. I'll capture in an issue.